src/android/MyConnectionService.java

```diff
@@ -1,65 +1,199 @@
         String payloadString = requestExtras.getString("payload");
+        Log.d(TAG, "onCreateIncomingConnection payload: " + payloadString);
+        JSONObject payload;
+        try {
+             payload = new JSONObject(payloadString);
+        } catch (JSONException e) {
+            throw new RuntimeException("Failed to parse payload string: " + e);
+        }
+
+        String _callUUID = null;
+        try {
+            _callUUID = payload.getString("call_uuid");
+        } catch (JSONException e) {
+            throw new RuntimeException("onCreateIncomingConnection no call uuid provided for this connection");
+        }
+        final String callUUID = _callUUID;
 
         final Connection connection = new Connection() {
+            CallNotification callNotification;
+
+            @Override
+            public void onShowIncomingCallUi() { // Only for self managed connections
+                Log.d(TAG, "onShowIncomingCallUi() invoked, for call_uuid: " + callUUID);
+                this.callNotification = new CallNotification(payloadString, context);
+                this.callNotification.show();
+            }
+
             @Override
             public void onAnswer() {
+                Log.d(TAG, "onAnswer()");
                 this.setActive();
-                // Intent intent = new Intent(CordovaCall.getCordova().getActivity().getApplicationContext(), CordovaCall.getCordova().getActivity().getClass());
-                // // intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK|Intent.FLAG_ACTIVITY_SINGLE_TOP);
-                // intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK|Intent.FLAG_ACTIVITY_SINGLE_TOP|Intent.FLAG_FROM_BACKGROUND);
-                // CordovaCall.getCordova().getActivity().getApplicationContext().startActivity(intent);
-                // ArrayList<CallbackContext> callbackContexts = CordovaCall.getCallbackContexts().get("answer");
-                // for (final CallbackContext callbackContext : callbackContexts) {
-                //     CordovaCall.getCordova().getThreadPool().execute(new Runnable() {
-                //         public void run() {
-                //             Bundle data = request.getExtras() != null ? request.getExtras() : new Bundle();
-                //             PluginResult result = new PluginResult(PluginResult.Status.OK, convertBundleToJson(data));
-                //             result.setKeepCallback(true);
-                //             callbackContext.sendPluginResult(result);
-                //         }
-                //     });
-                // }
-                // TelecomManager tm = (TelecomManager) CordovaCall.getCordova().getActivity().getApplicationContext().getSystemService(Context.TELECOM_SERVICE);
-                // tm.showInCallScreen(false);
+                activeConnection = this;
+                showWebApp();
 
                 // Allow enough time for our app to open and register the answer callback
```

Comment:
This needs to be solved a better way, rather than waiting an arbitrary period of time, we should probably emit the event when facetalk registers the listener to the "answer" event.

Will fix this up soon (and apply to all situations where were sending an event to facetalk)